CvGameCoreDLL_Expansion2/CvGame.cpp

```diff
@@ -9668,15 +9668,16 @@ void CvGame::updateMoves()
 
 		if(player.isAlive())
 		{
-			bool needsAIUpdate = player.hasUnitsThatNeedAIUpdate();
-			if(player.isTurnActive() || needsAIUpdate)
+			bool bAutomatedUnitNeedsUpdate = player.hasUnitsThatNeedAIUpdate();
+			bool bHomelandAINeedsUpdate = player.GetHomelandAI()->NeedsUpdate();
+			if(player.isTurnActive() || bAutomatedUnitNeedsUpdate || bHomelandAINeedsUpdate)
```

Comment:
This condition is repeated at least 3 times `|| bAutomatedUnitNeedsUpdate || bHomelandAINeedsUpdate`
Is there some `else` magic below so each if should have it?

Reply:
This code is a big old mess, it looks like it's pretty much unaltered from the original source code. Looks like there's a bunch of stuff that goes on in the different layers of if/else. From my testing it seems like this function is called once for each AI turn and continuously polled during the player's turn.

Reply:
For this change I needed to do a little addition to ensure that the homeland AI is updated even when the player has no automated units. (Otherwise worker recommendations won't show up).